runtime/src/bloom.rs

```diff
@@ -121,9 +122,59 @@ impl<T: AsRef<[u8]>> BloomHashIndex for T {
     }
 }
 
+pub struct AtomicBloom<T> {
+    num_bits: u64,
+    keys: Vec<u64>,
+    bits: Vec<AtomicU64>,
+    _phantom: PhantomData<T>,
+}
+
+impl<T: BloomHashIndex> From<Bloom<T>> for AtomicBloom<T> {
+    fn from(bloom: Bloom<T>) -> Self {
+        AtomicBloom {
+            num_bits: bloom.bits.len(),
+            keys: bloom.keys,
+            bits: bloom
+                .bits
+                .into_boxed_slice()
+                .iter()
+                .map(|&x| AtomicU64::new(x))
+                .collect(),
+            _phantom: PhantomData::default(),
+        }
+    }
+}
+
+impl<T: BloomHashIndex> AtomicBloom<T> {
+    pub fn add(&self, key: &T) {
+        for k in &self.keys {
+            let pos = key.hash_at_index(*k) % self.num_bits;
+            let index = pos >> 6;
```

Comment:
Just for future clarity, could we add some comments to these two lines, something like:

pos >> 6 divides by 64 to figure out which of the Atomic64 bit chunks we need to modify

(pos & 63) is equivalent to mod 63 so that we can find the index of the bit within the Atomic64 to modify

Reply:
done! `pos & 63` is equivalent to mod 64 though.

Reply:
ah yeah :)